services/nycrc/nycrc.service.js

```diff
@@ -0,0 +1,160 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { coveragePercentage } = require('../color-formatters')
+const {
+  ConditionalGithubAuthV3Service,
+} = require('../github/github-auth-service')
+const { fetchJsonFromRepo } = require('../github/github-common-fetch')
+const { InvalidParameter, InvalidResponse, NotFound } = require('..')
+
+const nycrcSchema = Joi.object({
+  branches: Joi.number()
+    .min(0)
+    .max(100),
+  lines: Joi.number()
+    .min(0)
+    .max(100),
+  functions: Joi.number()
+    .min(0)
+    .max(100),
+}).required()
+
+const pkgJSONSchema = Joi.object({
+  c8: Joi.object({
+    branches: Joi.number()
+      .min(0)
+      .max(100),
+    lines: Joi.number()
+      .min(0)
+      .max(100),
+    functions: Joi.number()
+      .min(0)
+      .max(100),
+  }).optional(),
+  nyc: Joi.object({
+    branches: Joi.number()
+      .min(0)
+      .max(100),
+    lines: Joi.number()
+      .min(0)
+      .max(100),
+    functions: Joi.number()
+      .min(0)
+      .max(100),
+  }).optional(),
+}).required()
+
+const documentation = `<p>
+  Create a code coverage badge, based on thresholds stored in a
+  <a href="https://github.com/istanbuljs/nyc#common-configuration-options">.nycrc config file</a>
+  on GitHub.
+</p>`
+
+const validThresholds = ['branches', 'lines', 'functions']
```

Comment:
@bcoe thank you for the badge, I'm curious if there was any reason not to add `'statements' here?